datafusion/physical-expr/src/equivalence.rs

```diff
@@ -869,9 +986,27 @@ impl EquivalenceProperties {
         self.ordering_satisfy_requirement(&sort_requirements)
     }
 
-    /// Checks whether the given sort requirements are satisfied by any of the
-    /// existing orderings.
+    /// Checks whether the given sort requirements are satisfied by any of the existing orderings.
+    /// This function applies an implicit projection to itself before calling `ordering_satisfy_requirement_helper`
+    /// to define the orderings of complex [`PhysicalExpr`]'s during analysis.
     pub fn ordering_satisfy_requirement(&self, reqs: LexRequirementRef) -> bool {
+        let exprs = reqs
+            .iter()
+            .map(|sort_req| sort_req.expr.clone())
+            .collect::<Vec<_>>();
+        let projection_mapping = self.implicit_projection_mapping(&exprs);
+        let projected_eqs =
+            self.project(&projection_mapping, projection_mapping.output_schema());
+        if let Some(projected_reqs) = projection_mapping.project_lex_reqs(reqs) {
```

Comment:
I am not quite sure I follow this logic -- does it say that any expression that can be projected maintains the ordering?

What about non monotonic expressions like `abs(x)` ? If the input is orderered by x

```
-2
-1
0
1
2
```
The output will not be ordered by `abs(xx)` 

```
2
1
0
1
2
```

(the same applies to date functions like `extract(day from ts)`

Perhaps we need to check [FuncMonotonicity](https://docs.rs/datafusion/latest/datafusion/physical_expr/functions/type.FuncMonotonicity.html#)

Reply:
> I am not quite sure I follow this logic -- does it say that any expression that can be projected maintains the ordering?
No, it just projects everything as if they are evaluated (in terms of ordering properties). For instance if `ts` is not ordered. We know that `date_bin(ts)` is not ordered. Similarly `abs(x)` is not ordered no matter `x` is ordered or not. `update_ordering` considers the properties of functions.

Reply:
I guess I was trying to say that in general, for `ProjectionExec` and any node that computes expressions, I would expect that we would have to use `FuncMonotonicity` to determine if an expression of `x` is ordered by `expr(x)`

Maybe that would be a good thing to do as a follow on PR